src/storage/CompositeStorage.cxx

```diff
@@ -21,13 +21,23 @@
 #include "CompositeStorage.hxx"
 #include "FileInfo.hxx"
 #include "fs/AllocatedPath.hxx"
+#include "fs/io/BufferedOutputStream.hxx"
+#include "storage/Registry.hxx"
```

Comment:
I'd like to keep this out of this class, because the way you implemented it, the generic `CompositeStorage` class now has a dependency on all storage plugins. `CompositeStorage` should stay generic and independent.

Reply:
I see. But to restore the mountpoints this dependency is neccessary. Having a `onRestoreMount` callback to move it out of this class only moves the dependency into `StateFile.cxx` which is even worse I think. I don't see any better solution right now. Could you give me a hint where to put it?

Reply:
Why is that worse? `StateFile.cxx` is glue code with a shitload of dependencies - to everything which can be referenced in the state file. That's a natural dependency. But I wouldn't have it exactly in `StateFile.cxx`, rather in a separate source like `StorageState.cxx` or so. It gets called from `StateFile.cxx` and calls into a `CompositeStorage` instance to save and restore its state.